src/objects.rs

```diff
@@ -371,7 +383,7 @@ pub unsafe trait HasEnergyForSpawn: CanStoreEnergy {}
 ///
 /// The reference returned from `AsRef<Reference>::as_ref` must be have a
 /// `cooldown` properties.
-pub unsafe trait HasCooldown: StructureProperties {
+pub unsafe trait HasCooldown: RoomObjectProperties {
```

Comment:
I think this is _technically_ a breaking change, so it probably wants a CHANGELOG entry?

Reply:
Good point, done!